banno-enrollment-po/BANNO.ENROLLMENT.V1.POW

```diff
@@ -0,0 +1,772 @@
+    END
+
+   ELSE IF ACTIONREQUESTED=ACTIONEDITXACCT OR
+           ACTIONREQUESTED=ACTIONADDXACCT THEN
+    DO
+     CALL SELECTPREF
+     CALL ADDEDITXACCTDIALOG
+     CALL ADDEDITXACCTFM
+     WHILE FMERROR<>"" 
+      DO
+       CALL ADDEDITXACCTDIALOG
+       CALL ADDEDITXACCTFM
+      END
+    END
+
+   ELSE IF ACTIONREQUESTED=ACTIONDELXACCT THEN
+    DO
+     CALL SELECTPREF
+     CALL DELETEXACCTDIALOG
+     CALL DELETEXACCTFM
+     WHILE FMERROR<>"" 
+      DO
+       CALL DELETEXACCTDIALOG
+       CALL DELETEXACCTFM
+      END
+    END
+
+   ELSE IF ACTIONREQUESTED=ACTIONEXITPGM THEN
+    DO
+     TERMINATE
+    END
+
+  END [ENDLESS LOOP]
+END [SETUP]
+
+PRINT TITLE="BANNO.ENROLLMENT.V1.POW"
+ SUPPRESSNEWLINE
+END [PRINT]
+
+PROCEDURE INITVARS
+[* Initializes variables before each run of specfile.
+*]
+ FOR PREFINDEX=0 TO PREFMAX
+  DO
+   PREFLOC(PREFINDEX)        = 0
+   PREFUSER(PREFINDEX)       = ""
+   PREFEMAILADDR(PREFINDEX)  = ""
+   PREFXFERANYLOC(PREFINDEX)     = 0  
+   PREFPASSPROB(PREFINDEX)       = 0
+  END
+ FOR ACCESSINDEX=0 TO ACCESSMAX
+  DO
+   ACCESSLOC(ACCESSINDEX)    = 0
+   ACCESSTYPE(ACCESSINDEX)   = 0
+   ACCESSACCT(ACCESSINDEX)   = ""
+   ACCESSIDTYPE(ACCESSINDEX) = 0
+   ACCESSID(ACCESSINDEX)     = ""
+   ACCESSWD(ACCESSINDEX)     = 0
+   ACCESSDEP(ACCESSINDEX)    = 0
+   ACCESSINQ(ACCESSINDEX)    = 0
+   ACCESSFM(ACCESSINDEX)     = 0
+  END
+ PREFCOUNT                   = 0
+ SELECTEDPREF                = 0
+ ACTIONREQUESTED             = 0
+ NEWPASSWORD                 = ""
+ NEWPASSPROB                 = ""
+ NEWPASSPROBNUM              = 0
+ ACCESSCOUNT                 = 0
+ SELECTEDACCESS              = 0
+ CREATEXFERTOANY             = ""
+ NEWUSERNAME                 = ""
+ NEWEMAILADDR                = ""
+ FMERROR                     = ""
+ NEWPREFLOC                  = 0
+ REALLYDELETE                = ""
+ NEWACCESSTYPE               = 0
+ NEWACCESSACCT               = ""
+ NEWACCESSIDTYPE             = 0
+ NEWACCESSID                 = ""
+ NEWACCESSWD                 = 0
+ NEWACCESSDEP                = 0
+ NEWACCESSINQ                = 0
+ NEWACCESSFM                 = 0
+ ACCESSDESC                  = ""
+END [PROCEDURE]
+
+PROCEDURE SELECTPREF
+[* Dialog that prompts user to choose banno username if more 
+** than one HB preference is present.
+*]
+ IF PREFCOUNT=1 THEN
+  SELECTEDPREF=1
+ ELSE IF PREFCOUNT>1 THEN
+  DO
+   DIALOGSTART("Select Banno User",200%,0)
+   DIALOGPROMPTLISTSTART("Choose Banno User",1)
+   FOR PREFINDEX=1 TO PREFCOUNT
+    DO
+     DIALOGPROMPTLISTOPTION(PREFINDEX,PREFUSER(PREFINDEX))
+   END
+   DIALOGPROMPTLISTEND
+   DIALOGDISPLAY
+   SELECTEDPREF=ENTERNUMBER("Choose Banno User",1)
+   DIALOGCLOSE
+  END
+END [PROCEDURE]
+
+PROCEDURE GETPREFERENCERECS
+[* Collects needed data from each HB preference on current
+** account.
+*]
+ PREFCOUNT=0
+ FOR EACH PREFERENCE WITH (PREFERENCE:HBUSERNAME<>"" AND
+                           PREFERENCE:HBMODE=STDHOMEBANKING)
+  DO
```

Comment:
Line 285: should you check the PREFCOUNT being less than PREFMAX before incrementing it by 1 to avoid an array out of bounds error?

Reply:
Done